scripts/newinstall.sh

```diff
@@ -557,47 +577,56 @@ git_check() {
 python_check() {
 	# Check the version by running a small Python program (taken from the Python
 	# EUPS package) XXX this will break if python is not in $PATH
-	local pyverok
-	pyverok=$(python -c 'import sys
-minver2=7
-minver3=5
-vmaj = sys.version_info[0]
-vmin = sys.version_info[1]
-if (vmaj == 2 and vmin >= minver2) or (vmaj == 3 and vmin >= minver3):
-    print(1)
-else:
-    print(0)')
-	if [[ $BATCH_FLAG = true ]]; then
-		WITH_MINICONDA=true
-	else
-		if [[ $pyverok != 1 ]]; then
+	local py_interp=python
+	local minver2=7
+	local minver3=5
+
+	local has_python=false
+	local pyverok=false
+	if n8l::has_cmd $py_interp; then
+		has_python=true
+		if n8l::pyverok $py_interp; then
+			pyverok=true
+		fi
+	fi
+
+	if [[ $pyverok != true ]]; then
+		if [[ $has_python == true ]]; then
 			cat <<-EOF
 
-			LSST stack requires Python 2 (>=2.7) or 3 (>=3.5); you seem to have
-			$(python -V 2>&1) on your path ($(which python)).  Please set up a
-			compatible python interpreter, prepend it to your PATH, and rerun this
-			script.  Alternatively, we can set up the Miniconda Python distribution
+			LSST stack requires Python 2 (>=2.${minver2}) or 3 (>=3.${minver3}); you
+			seem to have $($py_interp -V 2>&1) on your path ($(which $py_interp)).
+
+			EOF
+		else
+			cat <<-EOF
+
+			Unable to locate python.
+
+			Please set up a compatible python interpreter, prepend it to your PATH, and
+			rerun this script.  Alternatively, we can set up the Miniconda Python
+			distribution for you.
 
-			for you.
 			EOF
 		fi
+	fi
 
-		cat <<-EOF
+	cat <<-EOF
 
-		In addition to Python 2 (>=2.7) or 3 (>=3.5), some LSST packages depend on
-		recent versions of numpy, matplotlib, and scipy.  If you do not have all of
-		these, the installation may fail.  Using the Miniconda Python distribution
-		will ensure all these are set up.
+	In addition to Python 2 (>=2.${minver2}) or 3 (>=3.${minver3}), some LSST
+	packages depend on recent versions of numpy, matplotlib, and scipy.  If you
+	do not have all of these, the installation may fail.  Using the Miniconda
+	Python distribution will ensure all these are set up.
 
-		Miniconda Python installed by this installer will be managed by LSST\'s EUPS
-		package manager, and will not replace or modify your system python.
+	Miniconda Python installed by this installer will be managed by LSST\'s EUPS
+	package manager, and will not replace or modify your system python.
```

Comment:
I realise you're just moving the text above rather than changing it, but:

- I don't think you want to escape the `'` in `LSST's`;
- Is the bit about being managed by EUPS actually true? I believe you've changed that.